lib/internal/async_hooks.js

```diff
@@ -243,13 +245,84 @@ function restoreActiveHooks() {
   active_hooks.tmp_fields = null;
 }
 
+const triggerAsyncIdSymbol = Symbol('triggerAsyncId');
+const destroyedSymbol = Symbol('destroyed');
+
+function wrapPromise(promise, parent, silent) {
+  const asyncId = getOrSetAsyncId(promise);
+
+  if (parent) {
+    promise[triggerAsyncIdSymbol] = getOrSetAsyncId(parent);
+    promise.isChainedPromise = true;
```

Comment:
Not sure if it is ok to add an enumerable property one every promise. The Promise API is standardized.

Reply:
That's just there for API compatibility with the native wrapped version. I think we'll either need to add a non-enumerable copy here or remove the support from the native `PromiseWrap`. From previous discussions, it sounds like maybe no one is using `isChainedPromise` anymore?